### ISSUE TITLE
🐛 `internalError/io` when unknown files are passed as CLI arguments

### ISSUE THREAD

**Issue by Conaclos** [biomejs/biome#1493]
### Environment information

```block
CLI:
  Version:                      main branch
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           linux

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           unset
  JS_RUNTIME_NAME:              unset
  NODE_PACKAGE_MANAGER:         unset

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    false
  VCS disabled:                 false

Workspace:
  Open Documents:               0
```

### What happened?

```shell
❯ cat biome.json
{}

❯ ls src/
file1.js
file2.unknown

❯ ./biome format src/*
src/file2.unknown files/missingHandler ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ⚠ Biome doesn't know how to process this file
  

Compared 1 file(s) in 3ms
Skipped 1 file(s)
internalError/io ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ No files were processed in the specified paths.
```

### Expected result

No internal error should be emitted.

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by ematipico** [comment]
What's the bug? Biome doesn't support globs via command line

**Comment by Conaclos** [comment]
The glob is not passed to Biome. `./biome.fix format src/*` is expanded to `./biome.fix format src/file1.js src/file2.js` by the shell.
Thus, you get the same error without the glob:

```shell
❯ ./biome.fix format src/*
src/file2.unknown files/missingHandler ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ⚠ Biome doesn't know how to process this file
  

Compared 1 file(s) in 3ms
Skipped 1 file(s)
internalError/io ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ No files were processed in the specified paths.
```

The issue is the error.

**Comment by ematipico** [comment]
Thank you, I suspected so, but not every shell does glob expansion and we require a deterministic way to reproduce the issue.

**Comment by e965** [comment]
Faced this problem, but in a situation where we have svg files in the directory we are checking (I suppose this could be a fairly common case)

![image](https://github.com/biomejs/biome/assets/18071040/7cf4ae71-1708-47bc-8513-2b1fd16615f6)

**Comment by ematipico** [comment]
Your use case @e965 should be fixed in v1.5.1

**Comment by e965** [comment]
@ematipico Yeah, problem solved. Thank you!
